app/controllers/api/metrics/periodic_controller.rb

```diff
@@ -0,0 +1,16 @@
+class API::Metrics::PeriodicController < API::BaseController
+  def index
+    period_type = index_params.require(:period_type)
```

Comment:
You probably want to catch this and raise an `InvalidMetricPeriodError`, or not use `require` (ie require will raise a Strong Parameters exception if it's not passsed)